recipes/natural_language_processing/rag/app/rag_app.py

```
@@ -1,74 +1,61 @@
-vectorDB_client = HttpClient(host=vdb_host,
-                    port=vdb_port,
-                    settings=Settings(allow_reset=True,))
+def create_tmp_file():
+    with open(data.name, mode='wb') as w:
+        w.write(data.getvalue())
+    print(f"{data.name} uploaded")
 
-def clear_vdb():
-    global vectorDB_client
-    try:
-        vectorDB_client.delete_collection(vdb_name)
-        print("Cleared DB")
-    except:
-        pass
 
 st.title("📚 RAG DEMO")
 with st.sidebar:
-    data = st.file_uploader(label="📄 Upload Document",type=['txt'], on_change=clear_vdb)
+    data = st.file_uploader(label="📄 Upload Document",type=['txt'], on_change=vdb.clear_db)
 
 ### populate the DB ####
-os.environ["TOKENIZERS_PARALLELISM"] = "false"
-
-embedding_func = embedding_functions.SentenceTransformerEmbeddingFunction(model_name=embedding_model)
-e = SentenceTransformerEmbeddings(model_name=embedding_model)
-
-collection = vectorDB_client.get_or_create_collection(vdb_name,
-                                      embedding_function=embedding_func)
-if collection.count() < 1 and data != None:
-    print("populating db")
-    raw_documents = [Document(page_content=data.getvalue().decode("utf-8"),
-                              metadata={"":""})]
-    text_splitter = CharacterTextSplitter(separator = ".",
-                                          chunk_size=int(chunk_size),
-                                          chunk_overlap=0)
-    docs = text_splitter.split_documents(raw_documents) 
-    for doc in docs:
-        collection.add(
-            ids=[str(uuid.uuid1())],
-            metadatas=doc.metadata, 
-            documents=doc.page_content
-            )
-if data == None:
-    print("Empty VectorDB")
+if data != None:
+    with open(data.name, mode='wb') as w:
+        w.write(data.getvalue())
+    print(f"{data.name} uploaded")
+    documents = get_docs(data.name)
 else:
-    print("DB already populated")
+    documents = get_docs("../sample-data/fake_meeting.txt")
```

```suggestion

```
I don't think we want to reset `documents` to this hard coded value when `data == None`. I'd just print that the vdb is empty.

---

If `data == None` then what will the `retriever` object be? The app UI shows an error in the llm chain when it is none. However, once you upload the doc the error goes from the app. Do we want to keep that workflow? Or am I doing something wrong?

---

If there is no data, then the DB would be empty. The retriever would still get instantiated, just to an empty DB.  

I've not seen the error you are referencing.

In any case, I don't think defaulting to a hardcoded file name is the right solution.

---

I looked into instantiating the retriever to an empty DB, will update it. 
Agreed on the hardcoded file name, that is not a viable solution.